### ISSUE TITLE
Suggestions for chat improvement

### ISSUE THREAD

**Issue by suddenwhipvapor** [bisq-network/bisq2#1844]
There's a few things I think could be improved in the chat mechanics, and as suggested I am opening a specific issue about it.

 - allow direct text selection inside chat bubbles so users can then Ctrl-C; current behaviour has you click on three-dot menu to copy the whole message, which takes longer in some cases, and gets you all the text while you maybe are only looking for part of it
 - make links clickable (I understand this can be a security risk and was probably already addressed by not making links clickable, yet this could be a default behaviour that users can override in settings)
 - Shift-Enter just brings the cursor on the next line, ignoring if you press it with existing text after it; expected behaviour would be that the text after the cursor also gets pushed down one line
 - spice up the autoscroll feature: currently when you open a chat, it will be scrolled to about the place it was at the last time you opened it, with no clear indication of the last message you read, and sometimes it happens that you have to scroll all the way down from the top of the chat even if you had read it before; placing some sort of marker after the last read bubble, and moving the scroll there, would make it for support agents easier to give assistance
 - reply button doesn't focus the text field, so currently you have to click reply, and then click again to type
 - clicking on quoted message doesn't bring you to said message up in the history
 - in a nutshell, trying to replicate as much as possible Telegram chat mechanics would be the best approach, but the above bullets probably are those which would make the biggest difference for now

**Comment by HenrikJannsen** [comment]
> reply button doesn't focus the text field, so currently you have to click reply, and then click again to type

That is already fixed for 2.0.2.

**Comment by 54corbin** [comment]
I cant take this one.

**Comment by HenrikJannsen** [comment]
If any of the suggested improvements turn out to be overly complicate, lets discuss...

**Comment by 54corbin** [comment]
> allow direct text selection inside chat bubbles so users can then Ctrl-C; current behaviour has you click on three-dot menu to copy the whole message, which takes longer in some cases, and gets you all the text while you maybe are only looking for part of it

Hi @HenrikJannsen, regarding the text selection: Currently, chat messages use Labels, which are static and don't support direct selection.

To enable partial selection, the likely technical path is switching the underlying Label to a TextArea, with careful styling modifications (non-editable, transparent background/border, etc.) to match the current appearance.

Assuming this approach is acceptable, shall I proceed with implementing this change?

FYI:

![Image](https://github.com/user-attachments/assets/b963ede2-4a6d-4e1a-b2d4-cf4a41307f08)

**Comment by HenrikJannsen** [comment]
We have a component in place for such: `SelectableLabel` which swaps between a label and textArea.

Though I am a bit worries that it comes with a performance hit. The chat is anyway not super performant as it is, and we should check if adding a heavier component makes a detectable difference.

You can try to use the component, dont wast time on fine tuning but only try to check performance in chats with lot of messages and scroll up and down to see how it behaves. If you find some easy way to measure it would be good.
The we can decide if its worth it or not. Ppl can copy the text anyway, so I don't consider it a must-have but certainly a nice to have.

**Comment by 54corbin** [comment]
>allow direct text selection inside chat bubbles so users can then Ctrl-C; current behaviour has you click on three-dot menu to copy the whole message, which takes longer in some cases, and gets you all the text while you maybe are only looking for part of it

#3391

**Comment by 54corbin** [comment]
> clicking on quoted message doesn't bring you to said message up in the history

Seems it already implemented ?
![Image](https://github.com/user-attachments/assets/8ca43b46-747e-4cf8-a51c-24d361f6d9d6)

**Comment by 54corbin** [comment]
> Shift-Enter just brings the cursor on the next line, ignoring if you press it with existing text after it; expected behaviour would be that the text after the cursor also gets pushed down one line

#3392
